hypersync-client/src/decode.rs

```diff
@@ -18,33 +26,20 @@ impl Decoder {
     ///        "Transfer(address indexed from, address indexed to, uint amount)",
     ///     ]).unwrap();
     pub fn from_signatures<S: AsRef<str>>(signatures: &[S]) -> Result<Self> {
-        let mut map = signatures
+        let map: DecoderMap = signatures
             .iter()
             .map(|sig| {
                 let event =
                     alloy_json_abi::Event::parse(sig.as_ref()).context("parse event signature")?;
                 let topic0 = event.selector().to_vec();
+                let num_topics = event.num_topics();
+                let event_key = EventKey { topic0, num_topics };
                 let event = event.resolve().context("resolve event")?;
-                Ok((topic0, event))
+                Ok((event_key, event))
             })
-            .collect::<Result<Vec<_>>>()
+            .collect::<Result<DecoderMap>>()
             .context("construct event decoder map")?;
 
-        map.sort_unstable_by(|a, b| a.0.cmp(&b.0));
-
-        let initial_len = map.len();
-
-        map.dedup_by(|a, b| a.0 == b.0);
-
-        let new_len = map.len();
-
-        if initial_len != new_len {
-            return Err(anyhow!(
-                "duplicate event signature selectors (topic0) found.
-This might be because the 'indexed' keyword doesn't effect the selector of an event signature."
-            ));
```

Comment:
Removes the need for this code. Hashmap will deduplicate and account for different indexed keywords with same topic 0